src/App.js

```diff
@@ -2,46 +2,75 @@ import './styles/reset.css'
 import './styles/index.css'
 
 import initialStoreItems from './store-items'
-
-/*
-Here's what a store item should look like
-{
-  id: '001-beetroot',
-  name: 'beetroot',
-  price: 0.35
-}
-
-What should a cart item look like? 🤔
-*/
+import Store from './Store'
+import { useState } from 'react'
+import Cart from './Cart'
+import Total from './Total'
 
 console.log(initialStoreItems)
 
 export default function App() {
   // Setup state here...
+  const [cart, setCart] = useState([])
+  //const [quantity, setQuantity]= useState(1)
 
+  //  Adding item into Cart
+  function addToCart(item) {
+    // check if the item is already in the cart
+    // if it is increase the quantity by 1 and return
+    const itemFound = cart.find(itemFromCart => itemFromCart.id === item.id)
+    if (itemFound) {
+      const others = cart.filter(itemFromCart => itemFromCart.id !== item.id)
+      itemFound.quantity += 1
+      setCart([...others, itemFound])
+    } else {
+      const cartItem = { ...item, quantity: 1 }
+      setCart([...cart, cartItem])
+    }
+  }
+  // item increase  in the cart
+  const handleIncrement = item => {
+    const updatedCart = cart.map(cartItem => {
```

Comment:
Very nice, I think something like this for the above function would be great. Or maybe you can even reuse this code in the above function and just call it!